### ISSUE TITLE
2015-12-01 CaaC Standup

### ISSUE THREAD

**Issue by MatthewVita** [saxifrage/learn#362]
previous standup: https://github.com/saxifrage/cityasacampus/issues/353

What I accomplished last week:
- Worked with @whit537 on Heroku https://github.com/saxifrage/cityasacampus/pull/358
- Set up automated CaaC Development Environment Setup for Ubuntu Linux  https://github.com/saxifrage/cityasacampus/tree/master/dev

What I intend to do this week:
- I need to supply some feedback to @whit537 on the Heroku PR before we merge it (sorry for the delay on this, will get back to you after work today)
- See what I can work on wrt the dashboard

Obstacles that are impeding my progress:
- Nil

Favorite holiday song
- Dominick the Donkey

December availability
- I'll be around from now until the 24th and then will be back the 5th of next year (will probably sneak in bits of CaaC work here and there though)

**Comment by chadwhitacre** [comment]
last week:
- production deployment (#224, #351, #354, #358)

this week:
- dashboard angularization?
- actual real dashboard work? :-) "All hands on the Admin UI!"

blockers:
- none

song:
- "Come, thou long expected Jesus" (to Stuttgart)

December:
- planning on 4 (remote), 7, 11, 14, 18, 21, maybe 28

**Comment by dmtroyer** [comment]
Last week:
- Token auth #357 
- Reviewed and merged bulk add opportunities #341 

This week:
- Push up token auth for review, soon. I'm struggling to figure out where to cut it off. I think I'm just going to push up login functionality and then others can work off of that.
- User registration on angular.
- User password reset on angular.

Blockers:
- Myself.

Favorite Holiday Piece:
- A Ceremony of Carols by Benjamin Britten

December:
- I don't think my schedule will be that disrupted. Doing some traveling the week of Christmas but I may get some work done then, anyhow.

**Comment by chadwhitacre** [comment]
> Favorite holiday song

I suppose I missed an opportunity to pick something by the Grateful Dead. :-P

**Comment by chadwhitacre** [comment]
Since I wasn't able to come into the office today, I just called @timothyfcook on the phone. The impression I get is that we need to start bringing this project in for a landing. It seems that the "golden spike" will be when [the Rails app](http://learnpgh.org/) is connected to [the CaaC Map](http://caac-map.net/v1), so I'm going to turn my attention to that.

**Comment by chadwhitacre** [comment]
/me dives in on #255 and #227 ...

**Comment by chadwhitacre** [comment]
and #201 ... :swimmer:

**Comment by dmtroyer** [comment]
@whit537 just to confirm, you're doing #227 and #201 on the angular side, right?

**Comment by dmtroyer** [comment]
@whit537 @MatthewVita could I kindly request that merges to master be squashed first into more tangible feature groupings? Maybe one commit for a whole PR? I'm having some rebase issues and it makes it much more difficult to resolve them when I have no idea where the Heroku changes started, for example.

**Comment by chadwhitacre** [comment]
> could I kindly request that merges to master be squashed first into more tangible feature groupings? Maybe one commit for a whole PR?

Okay.

>  I'm having some rebase issues and it makes it much more difficult to resolve them when I have no idea where the Heroku changes started, for example.

Does the [commit list for the PR](https://github.com/saxifrage/cityasacampus/pull/358/commits) help?

**Comment by chadwhitacre** [comment]
> @whit537 just to confirm, you're doing #227 and #201 on the angular side, right?

Yeah, I'll branch from `token-auth` (#357).

**Comment by chadwhitacre** [comment]
I've toggled Travis to only build when `.travis.yml` is present. This will prevent the noisy build failures we're seeing on PRs since #361 hasn't landed yet, without hampering work on #361 itself.

![screen shot 2015-12-04 at 5 10 27 pm](https://cloud.githubusercontent.com/assets/134455/11602659/f4889760-9aa9-11e5-87bc-c824e1131d97.png)

**Comment by dmtroyer** [comment]
> I've toggled Travis to only build when .travis.yml is present. This will prevent the noisy build failures we're seeing on PRs since #361 hasn't landed yet, without hampering work on #361 itself.

Thank you @whit537 !

**Comment by dmtroyer** [comment]
> Does the commit list for the PR help?

Yes, it did! @whit537 Sorry to be such a grump.

**Comment by MatthewVita** [comment]
> Yes, it did! @whit537 Sorry to be such a grump.

{aside}
This is something I struggle with. I like to commit early/often along the way describing what the heck I'm doing but when merging feature xyz branch into master/main/trunk/etc, I don't know if squashing everything into a "adds feature xyz" commit is best or if keeping all of the commits help to "tell a story" in some way. I've seen it done both ways at my job and with other OSS projects. Anyways...
{aside}

**Comment by dmtroyer** [comment]
> This is something I struggle with. I like to commit early/often along the way describing what the heck I'm doing but when merging feature xyz branch into master/main/trunk/etc, I don't know if squashing everything into a "adds feature xyz" commit is best or if keeping all of the commits help to "tell a story" in some way. I've seen it done both ways at my job and with other OSS projects. Anyways...

The more I think about it, either way or both works for me. The tools are there to do what I was trying to do (diff before and after heroku), I was just having a crotchety knee-jerk reaction to the situation.

**Comment by chadwhitacre** [comment]
> I was just having a crotchety knee-jerk reaction to the situation.

:relaxed: 

> I've seen it done both ways at my job and with other OSS projects.

Ditto. Seems that squashing is a little more work at merge time for less work when reading history later on. I generally find that I merge PRs much more frequently than I read history, and consequently have preferred to optimize that step (i.e., by _not_ squashing commits). I tend to view commit squashing as a sign of a mature, stable project, largely because [a couple years ago](https://botbot.me/freenode/gratipay/2013-09-03/?msg=5683492&page=3) I learned that jQuery maintains an extremely clean history:

![screen shot 2013-09-02 at 11 33 41 pm](https://cloud.githubusercontent.com/assets/134455/11605583/9c357268-9ad1-11e5-9301-690c06fafef9.png)

On the other hand, I learned in the same conversation that [GitHub doesn't](https://twitter.com/holman/status/374732583122128897). :shrug:

> keeping all of the commits help to "tell a story" in some way

GitHub keeps all the old commits around even after a rebase-n-squash. If after rebasing you reference the previous head and/or the list of commits, then someone who wants to see the detailed story of the PR is still more or less able to. Here's [a recent example](https://github.com/vigetlabs/gulp-rails-pipeline/pull/40#issuecomment-160680963).

I've actually never worked regularly on a project that observes commit squashing, so I'd be fine to try it out here, especially if we can observe the practice of linking to the previous commits list as a way to preserve full history.
